hasher-matcher-actioner/terraform/indexer/main.tf

```diff
@@ -1,5 +1,72 @@
 # Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
 
+# First define the Indexing Schedule
+resource "aws_cloudwatch_event_rule" "indexing_schedule" {
+  name                = "${var.prefix}-HMA_IndexingSchedule"
+  description         = "Rebuild Index on a regular cadence"
+  schedule_expression = "rate(${var.indexer_frequency})"
+  role_arn            = aws_iam_role.indexing_schedule.arn
+}
+
+resource "aws_iam_role" "indexing_schedule" {
+  name_prefix        = "${var.prefix}_indexing_schedule"
+  assume_role_policy = data.aws_iam_policy_document.indexing_schedule_assume_role.json
+
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "FetcherLambdaTriggerRole"
+    }
+  )
+}
+
+data "aws_iam_policy_document" "indexing_schedule_assume_role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["events.amazonaws.com"]
+    }
+  }
+}
+
+## Define a policy document to asign to the role
+data "aws_iam_policy_document" "indexing_schedule" {
+  statement {
+    actions   = ["lambda:InvokeFunction"]
+    resources = ["*"]
```

Comment:
seems a tad dubious to allow the trigger/scheduler to invoke all lambdas, can we put the unified indexer lambda arn here? (I know the issue also exists for the fetcher)